pkg/resources/reconciling/modifier/related_revisions.go

```diff
@@ -46,6 +46,12 @@ func RelatedRevisionsLabels(ctx context.Context, client ctrlruntimeclient.Client
 	}
 }
 
+// AddRevisionLabelsToObject is a helper function that adds revisions of the resources that the given object depends on, to the given object labels.
+// The revision labels are used to trigger a restart of the object when the dependent resources are updated.
+func AddRevisionLabelsToObject(ctx context.Context, client ctrlruntimeclient.Client, obj ctrlruntimeclient.Object) (ctrlruntimeclient.Object, error) {
+	return addRevisionLabelsToObject(ctx, client, obj)
+}
+
```

Comment:
Why not export the internal function directly?

Reply:
I just did not want to update callers of the internal method; instead created a public one that calls directly the internal one. But if you think this is redundant, I can update the code accordingly.